### PR TITLE
hwmv2: Port STM32L4X SoCs & boards

### PR DESCRIPTION
**Pull request by gautierg-st**
Port STM32L4X SoCs & boards to Hardware Model V2.

### CODE REVIEW

File: boards/st/nucleo_l452re/board.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: nucleo_l452re
+  vendor: st
+  socs:
+    - name: stm32l452xx
```
Comment (erwango) [inline boards/st/nucleo_l452re/board.yml:5]: Missing variant to indicate p variant is available

File: boards/st/nucleo_l452re/nucleo_l452re_p.dts
Comment (erwango) [inline boards/st/nucleo_l452re/nucleo_l452re_p.dts:1]: should be renamaed as nucleo_l452re_stm32l45xx_p_defconfig, same for dts and yaml file

File: boards/st/nucleo_l452re/Kconfig.nucleo_l452re [hunk 2]
```diff
@@ -3,3 +3,6 @@
 
 config BOARD_NUCLEO_L452RE
 	select SOC_STM32L452XX
+
+config BOARD_NUCLEO_L452RE_STM32L452XX_P
+	select SOC_STM32L452XX
```
Comment (erwango) [inline boards/st/nucleo_l452re/Kconfig.nucleo_l452re:8]: Does this work ?
From what I've seen in other cases, it was required to indicate symbol type (bool)

**Review by erwango — CHANGES_REQUESTED** [review]
Please update _defconfig files path in boards doc

File: boards/st/nucleo_l452re/board.yml (hunk 1 above)
Comment (gautierg-st) [reply boards/st/nucleo_l452re/board.yml:5]: Done

File: boards/st/nucleo_l452re/nucleo_l452re_p.dts
Comment (gautierg-st) [reply boards/st/nucleo_l452re/nucleo_l452re_p.dts:1]: Done

File: boards/st/nucleo_l452re/Kconfig.nucleo_l452re (hunk 2 above)
Comment (gautierg-st) [reply boards/st/nucleo_l452re/Kconfig.nucleo_l452re:9]: Done

**Review by ajarmouni-st — APPROVED** [review]

File: boards/st/nucleo_l452re/Kconfig.defconfig [hunk 3]
```diff
@@ -6,10 +6,6 @@
 
 if BOARD_NUCLEO_L452RE || BOARD_NUCLEO_L452RE_P
```
Comment (erwango) [inline boards/st/nucleo_l452re/Kconfig.defconfig]: if BOARD_NUCLEO_L452RE

File: boards/st/nucleo_l452re/Kconfig.nucleo_l452re (hunk 2 above)
Comment (erwango) [reply boards/st/nucleo_l452re/Kconfig.nucleo_l452re:9]: Late finding: `	select SOC_STM32L452XX` should be removed.
From what I understand `BOARD_NUCLEO_L452RE` is set to y for both configurations.
Note that `BOARD_NUCLEO_L452RE_STM32L452XX_P` may not be needed at all if not used. For instance in no Kconfig difference exist.

**Review by erwango — CHANGES_REQUESTED** [review]
Please also update files paths in documentation (for instance boards/arm/swan_r5/swan_r5_defconfig)

File: boards/st/nucleo_l452re/Kconfig.nucleo_l452re (hunk 2 above)
Comment (gautierg-st) [reply boards/st/nucleo_l452re/Kconfig.nucleo_l452re]: Done

**Review by erwango — APPROVED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
Please fix adi_eval_adin1110ebz and adi_eval_adin1110ebz. Missing migration ?

Rebase needed

**Comment by gautierg-st** [comment]
> Please fix adi_eval_adin1110ebz and adi_eval_adin1110ebz. Missing migration ?
> 
> Rebase needed

For some reason they weren't migrated first time around. I re-ran the script and it worked this time.

File: boards/adi/adi_eval_adin1110ebz/doc/index.rst [hunk 4]
```diff
@@ -82,7 +82,7 @@ following hardware features:
 
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/adi_eval_adin1110ebz/adi_eval_adin1110ebz_defconfig``
+	``boards/adi/adi_eval_adin1110ebz/adi_eval_adin1110ebz_defconfig``
```
Comment (erwango) [inline boards/adi/adi_eval_adin1110ebz/doc/index.rst:85]: :zephyr_file:`boards/adi/adi_eval_adin1110ebz/adi_eval_adin1110ebz_defconfig`. Could be done in a single commit.

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by erwango — APPROVED** [review]

File: boards/st/nucleo_l452re/Kconfig.nucleo_l452re [hunk 5]
```diff
@@ -3,3 +3,6 @@
 
 config BOARD_NUCLEO_L452RE
 	select SOC_STM32L452XX
+
+config BOARD_NUCLEO_L452RE_STM32L452XX_P
+	bool
```
Comment (nordicjm) [inline boards/st/nucleo_l452re/Kconfig.nucleo_l452re:8]:
```suggestion
```

File: boards/adi/adi_eval_adin1110ebz/Kconfig.adi_eval_adin1110ebz [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2024 BayLibre
```
Comment (nordicjm) [inline boards/adi/adi_eval_adin1110ebz/Kconfig.adi_eval_adin1110ebz:1]: revert image change, the image size error is to be ignored, reducing an image size actually increases the repo size

**Review by nordicjm — DISMISSED** [review]

File: boards/st/nucleo_l452re/Kconfig.nucleo_l452re (hunk 5 above)
Comment (gautierg-st) [reply boards/st/nucleo_l452re/Kconfig.nucleo_l452re]: Done

File: boards/adi/adi_eval_adin1110ebz/Kconfig.adi_eval_adin1110ebz (hunk 6 above)
Comment (gautierg-st) [reply boards/adi/adi_eval_adin1110ebz/Kconfig.adi_eval_adin1110ebz:1]: Done

**Review by gmarull — APPROVED** [review]

**Pull request merged**